portal/models/identifier.py

```diff
@@ -48,9 +48,15 @@ def from_fhir(cls, data):
     def __str__(self):
         return 'Identifier {0.use} {0.system} {0.value}'.format(self)
 
+    def __key(self):
+        # Only use (system, value), as per unique constraint
+        return (self.system, self.value)
```

Comment:
Can you add `self.id` to the beginning of that tuple?

Reply:
Previously `__eq__` used only `self.system` and `self.value` and had that comment `Use only (system, value) as per unique constraint` - which makes me think that either id can change or there's another reason to not use it in `__eq__`, and therefore to not use it in `__key()`

Reply:
Good point!
@pbugni Maybe we should only check `self.id` in `__hash__()`? eg:

```
def __hash__(self):
  return hash((self.id,) + self.__key())
```

Reply:
no no - don't go looking for `id` - as that is a primary key set *after* `db.session.commit()`.
We use this, for example, when looking up identifiers from site persistence, they may be generated on the fly from a JSON file, and not yet committed.  
@John0Randolph 's impl of `__key__` looks spot on to me.